pkg/cluster/px/px.go

```diff
@@ -524,21 +528,28 @@ func (ops *pxClusterOps) upgradePX(newVersion string) error {
 			return nil, true, errNoPXDaemonset
 		}
 
+		pxImage := fmt.Sprintf("%s:%s", spec.PXImage, spec.PXTag)
+		ociImage := fmt.Sprintf("%s:%s", spec.OCIMonImage, spec.OCIMonTag)
+
 		for _, ds := range dss {
 			skip := false
 			logrus.Infof("Upgrading PX daemonset: [%s] %s to version: %s", ds.Namespace, ds.Name, newVersion)
 			dsCopy := ds.DeepCopy()
 			for i := 0; i < len(dsCopy.Spec.Template.Spec.Containers); i++ {
 				c := &dsCopy.Spec.Template.Spec.Containers[i]
 				if c.Name == pxDaemonsetContainerName {
-					if c.Image == newVersion {
+					oldPxImage := getEnv(c.Env, pxImageEnvKey)
+					if c.Image == ociImage && (spec.OCIMonTag == spec.PXTag || oldPxImage == pxImage) {
```

Comment:
this will fail in this scenario

daemonset running portworx/oci-monitor:2.0.3.3, it has PX_IMAGE pointing to portworx/px-enteprise:2.0.3.1

User requests talisman upgrade with pximage and tag portworx/px-enteprise:2.0.3.3.

In ` (spec.OCIMonTag == spec.PXTag || oldPxImage == pxImage)`, the first check will pass and since it's OR, it will skip the upgrade.